### ISSUE TITLE
default css length units should be "px", not "none"

### ISSUE THREAD

**Issue by mgambrell** [litehtml/litehtml#234]
Take this html for instance.
```
<html>
<head><style>span.bigwords { font-size: 20 } span.smallwords { font-size: 4 } </style></head>
<body>
<span class="bigwords">BIG WORDS.<br></span><span class="smallwords">WORDS. MORE WORDS.</span>
</body></html>
```
Eventually, I realized I should have set e.g. `font-size: 20px` and then things will work. It seems both chrome and firefox assume units of px. While litehtml has the units "none", the CSS specifications do not seem to define it. Actually the specifications seem to mandate the use of a unit; therefore my use of and firefox/chrome's support of unitless lengths are undefined behavior. Litehtml is generally using `css_units_none` to specify an "unset" length... i guess, kind of as if it wasn't there at all. This should probably not be changed; however, when a nice-looking (unitless) number is parsed as a length, I think at that moment it should be interpreted as pixels instead. So, for instance, css_length.cpp's `css_length::fromString` should change this:
`m_units = (css_units) value_index(un, css_units_strings, css_units_none);`
To this:
`m_units = (css_units) value_index(un, css_units_strings, css_units_px);`

Arguably this is a poor choice, since the generic tool to parse a css length should not bake-in such a cryptic semantic rule. So you could kick it up one more level perhaps and say "if units none, units shall be px instead". However, you will have to do this in a number of places. Perhaps `css_length::fromString` could take another parameter to indicate whether it's subject to these semantic corrections (or another function `css_length::fromStringLenient` could be added to wrap fromString and adjust `none` to `px`)

**Comment by tordex** [comment]
Actually ```css_units_none``` is not used to specify unset length and in most cases ```css_units_none``` is equal ```px```, see the ```document::to_pixels``` usage. With font-size is just not removed legacy code. Fixed in commit ae9fe10.

**Comment by mgambrell** [comment]
thank you!!
